parameter/ConfigurableDomain.cpp

```diff
@@ -769,7 +769,10 @@ bool CConfigurableDomain::getElementSequence(const string& strConfiguration, str
     return true;
 }
 
-bool CConfigurableDomain::setApplicationRule(const string& strConfiguration, const string& strApplicationRule, const CSelectionCriteriaDefinition* pSelectionCriteriaDefinition, string& strError)
+bool CConfigurableDomain::setApplicationRule(const string& strConfiguration,
+                                             const string& strApplicationRule,
+                                             const core::selection::criterion::Criteria& criteria,
```

Comment:
The 'selection' namespace doesn't seem very useful yet: everything it contains is contained is its sub-namespace 'criterion'.